java/ql/src/Metrics/Summaries/LinesOfCode.ql

```diff
@@ -0,0 +1,13 @@
+/**
+ * @id java/summary/lines-of-code
+ * @name Total lines of code in the database
+ * @description The total number of lines of code across all files. This is a useful metric of the size of a database.
+ *              For all files that were seen during the build, this query counts the lines of code, excluding whitespace
+ *              or comments.
+ * @kind metric
+ * @tags summary
+ */
+
+import java
+
+select sum(File f, int i | i = f.getNumberOfLinesOfCode() | i)
```

Comment:
Assuming that `getNumberOfLinesOfCode` is functional, this simplifies to
```suggestion
select sum(File f | f.getNumberOfLinesOfCode())
```

Reply:
I think I tried this, and I got an error along the lines of "sum can only be applied to int or float".

Reply:
Ah, I guess you'd need an extra `|`, then:
```suggestion
select sum(File f | | f.getNumberOfLinesOfCode())
```

Reply:
We probably want to exclude counting "code" in XML files:
```suggestion
select sum(CompilationUnit f | f.fromSource() | f.getNumberOfLinesOfCode())
```